src/Umbraco.Web/PublishedCache/NuCache/ContentCache.cs

```diff
@@ -29,194 +25,12 @@ internal class ContentCache : PublishedCacheBase, IPublishedContentCache2, INavi
         // it's too late for UmbracoContext which has captured previewDefault and stuff into these ctor vars
         // but, no, UmbracoContext returns snapshot.Content which comes from elements SO a resync should create a new cache
 
-        public ContentCache(bool previewDefault, ContentStore.Snapshot snapshot, IAppCache snapshotCache, IAppCache elementsCache, IDomainCache domainCache, IGlobalSettings globalSettings, IVariationContextAccessor variationContextAccessor)
+        public ContentCache(bool previewDefault, ContentStore.Snapshot snapshot, IVariationContextAccessor variationContextAccessor)
             : base(previewDefault)
         {
             _snapshot = snapshot;
-            _snapshotCache = snapshotCache;
-            _elementsCache = elementsCache;
-            _domainCache = domainCache;
-            _globalSettings = globalSettings;
             _variationContextAccessor = variationContextAccessor;
         }
-
-        private bool HideTopLevelNodeFromPath => _globalSettings.HideTopLevelNodeFromPath;
-
-        #endregion
-
-        #region Routes
-
-        // routes can be
-        // "/"
-        // "123/"
-        // "/path/to/node"
-        // "123/path/to/node"
-
-        // at the moment we try our best to be backward compatible, but really,
-        // should get rid of hideTopLevelNode and other oddities entirely, eventually
-
-        public IPublishedContent GetByRoute(string route, bool? hideTopLevelNode = null, string culture = null)
-        {
-            return GetByRoute(PreviewDefault, route, hideTopLevelNode, culture);
-        }
-
-        public IPublishedContent GetByRoute(bool preview, string route, bool? hideTopLevelNode = null, string culture = null)
-        {
-            if (route == null) throw new ArgumentNullException(nameof(route));
-
-            var cache = preview == false || PublishedSnapshotService.FullCacheWhenPreviewing ? _elementsCache : _snapshotCache;
-            var key = CacheKeys.ContentCacheContentByRoute(route, preview, culture);
```

Comment:
The usage of `CacheKeys.ContentCacheContentByRoute` seems to be removed in the refactoring. Was this intended? Do we still use this cache key?

Reply:
No longer used. CacheKeys.ContentCacheContentIdByRoute is used instead.

Reply:
Need to reimplement this cache somewhere now that CacheKeys.ContentCacheContentIdByRoute is removed

Reply:
Implemented in ContentCacheAdapter